dask_cuda/explicit_comms/comms.py

```diff
@@ -52,11 +59,31 @@ def default_comms(client: Optional[Client] = None) -> "CommsContext":
     -------
     comms: CommsContext
         The default comms object
+
+    Notes
+    -----
+    There are some subtle points around explicit-comms and the lifecycle
+    of a Dask Cluster.
+
+    A :class:`CommsContext` establishes explicit communication channels
+    between the workers *at the time it's created*. If workers are added
+    or removed, they will not be included in the communication channels
+    with the other workers.
+
+    If you need to refresh the explicit communications channels, then
+    create a new :class:`CommsContext` object or call ``default_comms``
+    again after workers have been added to or removed from the cluster.
     """
-    global _default_comms
```

Comment:
Python question - Do you happen to know why we needed `global` before, but don't need it anymore?

Reply:
I believe it's whether you're you're reassigning the object or not. Previously, we changed the name `_default_comms` from referencing `None` to some `CommsContext`. Now the object `_comms_cache` isn't being reassigned. It's always that WeakValueDictionary. We just happen to mutate it in this function.